### ISSUE TITLE
Create pre-flight checks for Elasticsearch and MongoDB versions

### ISSUE THREAD

**Issue by bernd** [Graylog2/graylog2-server#5248]
Graylog is currently not doing early pre-flight checks for required versions of Elasticsearch and MongoDB.

We only rely on driver compatibility checks and some manual checks in some parts of our code base at the moment. This might cause problems when users are upgrading to a new Graylog release without upgrading required services like Elasticsearch or MongoDB.
Especially using the wrong Elasticsearch version can break Graylog setups including data loss.

To avoid any problems, Graylog should run very early pre-flight checks for required services. These checks should run before we start any internal services, probably even before we create the dependency injector instance.
If one of these pre-flight checks fails, the Graylog process should be stopped and the user be notified about the problems.

## Proposed Checks & Tasks

- Check Elasticsearch version
- Check MongoDB version
- Check if there is enough space for the message journal and maybe check that the journal is not on a network filesystem like NFS (if that's possible)
- Run database migrations (pure MongoDB client ones) (see https://github.com/Graylog2/graylog2-server/issues/6671)
- Run ES index mapping migrations

**Comment by bernd** [comment]
Refs https://github.com/Graylog2/graylog-plugin-enterprise/issues/2966

**Comment by tellistone** [comment]
Jusr seen another customer accidentally upgraded to ES (7.16)

HS-703690514

**Comment by mpfz0r** [comment]
Regarding the ES version checks see Kay's comment over here: https://github.com/Graylog2/graylog-plugin-enterprise/issues/2966#issuecomment-981769841
tldr: doesn't make much sense to prevent Graylog from starting with an incompatible ES version.

**Comment by bernd** [comment]
> Regarding the ES version checks see Kay's comment over here: [Graylog2/graylog-plugin-enterprise#2966 (comment)](https://github.com/Graylog2/graylog-plugin-enterprise/issues/2966#issuecomment-981769841) tldr: It doesn't make sense to prevent Graylog from starting with ES version > 7.10

When users are upgrading, the startup check doesn't help much. I agree. But when people start new setups, it would prevent them, from starting with an unsupported version.

**Comment by mpfz0r** [comment]
> @mpfz0r When users are upgrading, the startup check doesn't help much. I agree. But when people start new setups, it would prevent them from starting with an unsupported version.

Makes sense. I will add that once I have "fresh install" detection ready :+1:

**Comment by boosty** [comment]
Great work! 🎉 

I'll assign @dulanism on this one and the related PR to handle the documentation.
